src/main/java/seedu/duke/Parser.java

```diff
@@ -102,30 +132,57 @@ private static String getProcessedAdd(String userInput, FoodList foodList) throw
      * @throws DietException when the user input is of a wrong format.
      */
     private static void executeProcessedInfo(String userInput, Manager manager) throws DietException {
-        Gender gender;
-        ActivityLevel actLvl;
-        String[] processedParam = getCommandParam(userInput).split(" ");
-        String processGender = processedParam[0].substring(processedParam[0].indexOf("/") + 1);
-        if (processGender.equals("M")) {
-            gender = Gender.MALE;
-        } else {
-            gender = Gender.FEMALE;
-        }
-        int age = Integer.parseInt(processedParam[1].substring(processedParam[1].indexOf("/") + 1));
-        int height = Integer.parseInt(processedParam[2].substring(processedParam[2].indexOf("/") + 1));
-        int orgWeight = Integer.parseInt(processedParam[3].substring(processedParam[3].indexOf("/") + 1));
-        int tarWeight = Integer.parseInt(processedParam[4].substring(processedParam[4].indexOf("/") + 1));
-        String processActLvl = processedParam[5].substring(processedParam[5].indexOf("/") + 1);
-        if (processActLvl.equals("1")) {
-            actLvl = ActivityLevel.NONE;
-        } else if (processActLvl.equals("2")) {
-            actLvl = ActivityLevel.LOW;
-        } else if (processActLvl.equals("3")) {
-            actLvl = ActivityLevel.MEDIUM;
-        } else if (processActLvl.equals("4")) {
-            actLvl = ActivityLevel.HIGH;
-        } else {
-            actLvl = ActivityLevel.EXTREME;
+        Gender gender = Gender.MALE;
+        ActivityLevel actLvl = ActivityLevel.NONE;
+        int age = 0;
+        int height = 0;
+        int orgWeight = 0;
+        int tarWeight = 0;
+        String trimmedParam;
+        String[] processedParam;
+        String[] paramList = {"g/", "a/", "h/", "o/", "t/", "l/"};
+        for (String param: paramList) {
+            processedParam = getCommandParam(userInput).split(param);
+            trimmedParam = processedParam[1].trim();
+            if (processedParam[1].contains("/")) {
+                trimmedParam = processedParam[1].substring(0, processedParam[1].indexOf("/") - 2).trim();
+            }
+            switch (param) {
+            case "g/":
+                String processGender = trimmedParam;
+                if (processGender.equals("M")) {
+                    gender = Gender.MALE;
```

Comment:
Perhaps consider changing female to an if-else case and the else case can throw an exception. Maybe you would like to consider this for activity level as well.